cli/flox-rust-sdk/src/models/manifest/composite/mod.rs

```diff
@@ -24,32 +24,113 @@ use super::typed::{
+    fn from_iter<T: IntoIterator<Item = Key>>(iter: T) -> Self {
+        Self(iter.into_iter().map(|k| k.into()).collect())
+    }
+}
+
+/// A warning that occurred during the merge of two manifests.
+/// This is used to provide feedback to the user about potential issues.
+///
+/// Warnings are not errors, but they may indicate
+/// that the user should review the merged manifest or its dependencies.
+///
+/// Currently, the only warning is that a value is being overridden,
+/// but more warnings may be added in the future.
+#[derive(Debug, Clone, PartialEq)]
+#[must_use]
+pub enum Warning {
+    Overriding(KeyPath),
+}
+
+/// A warning that occurred during the merge of two manifests,
+/// along with the names of the two manifests involved.
+#[derive(Debug, Clone, PartialEq)]
+pub struct WarningWithContext {
+    warning: Warning,
+    lower_priority_name: String,
+    higher_priority_name: String,
+}
+
 /// A collection of manifests to be merged with a `ManifestMergeStrategy`.
 #[derive(Debug, Clone, Default)]
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 struct CompositeManifest {
     composer: Manifest,
     #[cfg_attr(
         test,
-        proptest(strategy = "proptest::collection::vec(any::<Manifest>(), 0..=2)")
+        proptest(strategy = "proptest::collection::vec(any::<(String, Manifest)>(), 0..=2)")
     )]
-    deps: Vec<Manifest>,
+    deps: Vec<(String, Manifest)>,
 }
 
 impl CompositeManifest {
-    fn merge_all(&self, merger: impl ManifestMergeStrategy) -> Result<Manifest, MergeError> {
-        let Some(first_dep) = self.deps.first() else {
-            // No deps, just composer.
-            return Ok(self.composer.clone());
-        };
+    fn merge_all(
+        &self,
+        merger: impl ManifestMergeStrategy,
+    ) -> Result<(Manifest, Vec<WarningWithContext>), MergeError> {
+        let current_manifest = &("Current manifest".to_string(), self.composer.clone());
 
-        self.deps
-            .iter()
-            .skip(1) // First dep is used as initializer.
-            .chain(once(&self.composer)) // Composer goes last.
-            .try_fold(first_dep.clone(), |merged, next| {
-                merger.merge(&merged, next)
-            })
+        let mut merges = self.deps.iter().chain([current_manifest]);
+        let (mut prev_manifest_id, mut merged_manifest) = merges
+            .next()
+            .expect("including composer, there should be at least one manifest")
+            .clone();
+
+        let mut warnings = Vec::new();
+
+        for (manifest_id, manifest) in merges {
+            let (merged, merge_warnings) = merger.merge(&merged_manifest, manifest)?;
+            merged_manifest = merged;
+            let merge_warnings = merge_warnings
+                .into_iter()
+                .map(|warnings| WarningWithContext {
+                    warning: warnings,
+                    lower_priority_name: prev_manifest_id.clone(),
```

Comment:
Echoing your own suggestion in Slack, since `lower_priority_name` isn't accurate, I think we can probably just drop it for now